### DISCUSSION TITLE
React component disappears after Livewire refresh

### DISCUSSION THREAD

**Discussion by lucaguindani** [livewire/livewire#6680] [1. Help]
Hello,

I'm trying to make a simple Livewire (v3) component with a react component inside the blade view.

When I load the page the react component (a media player -> id="rct-player") will render without any issue, but when the Livewire component will refresh (wire:poll), the react component won't be re-rendered and will juste disappear from the view (no console errors ou anything else). I'm probably missing something here, but Is it possible to re-render a react component inside a view when the Livewire component is refreshed ? This will be used to automatically show a media player to the users when the uploaded media is processed/transcoded.

```
<?php

namespace App\Livewire;

use App\Card;
use Livewire\Component;

class Player extends Component
{
    public Card $card;

    public string $reference;

    public function mount(Card $card, string $reference)
    {
        $this->card = $card;
        $this->reference = $reference;
    }

    public function render()
    {
        return view('livewire.player');
    }
}
```
```
<div class="media-player" {{ $card->boxIsEditable($reference) ? 'wire:poll' : '' }}>
    @if(Helpers::hasSource($card))
        @if(Helpers::hasExternalLink($card))
            <div id="rct-player"
                 data='{{ json_encode(['locale' => Helpers::currentLocal(), 'card' => $card, 'url' => Helpers::getExternalLink($card), 'isLocal' => false]) }}'
            ></div>
        @elseif($card->file)
            @if(Helpers::isFileStatus($card->file, \App\Enums\FileStatus::Ready))
                <div id="rct-player"
                     data='{{ json_encode(['locale' => Helpers::currentLocal(), 'card' => $card, 'url' => Helpers::fileUrl($card->file->filename), 'isLocal' => true]) }}'
                ></div>
            @elseif(Helpers::isFileStatus($card->file, \App\Enums\FileStatus::Failed))
                <p class="text-danger text-center fs-5 lh-sm p-3">
                    {{ trans('messages.card.media.failed') }}
                </p>
            @else
                <p class="text-secondary text-center fs-5 lh-sm p-3">
                    {{ trans('messages.card.media.processing') }}
                </p>
            @endif
        @endif
    @else
        <p class="text-secondary text-center p-4">
            {{ trans('messages.card.media.not.selected') }}
        </p>
    @endif
</div>
```

Many thanks

**Comment by devhau** [comment]
Hey,
You can use wire:key

`<div class="media-player" {{ $card->boxIsEditable($reference) ? 'wire:poll' : '' }}>`

=>

`<div class="media-player" wire:key="card-{{$card->boxIsEditable($reference) ?1:0}}" {{ $card->boxIsEditable($reference) ? 'wire:poll' : '' }}>`

i try working it.

**Reply by lucaguindani** [reply]
Thanks for your help @devhau. I tried with the `wire:key` but the react component "rct-player" is still disappearing (not rendered) on the first auto-refresh (poll).

**Reply by devhau** [reply]
I think you try call refresh parent component

**Reply by lucaguindani** [reply]
What do you call the parent component ? I have a Livewire component with a React component inside the Livewire component blade view. When the Livewire component is refreshed, then the React component is not be re-rendered and will disappear from the view.

Even with an even simpler Livewire component I can recreate the issue (example below).

```
<div>
    <div id="rct-player"
         data='{{ json_encode(['locale' => Helpers::currentLocal(), 'card' => $card, 'url' => Helpers::fileUrl($card->file->filename), 'isLocal' => true]) }}'
    ></div>
    <button wire:click="$refresh">Refresh</button>
</div>
```

```
<?php

namespace App\Livewire;

use App\Card;
use Livewire\Component;

class Player extends Component
{
    public Card $card;

    protected $listeners = ['refreshComponent' => '$refresh'];

    public function mount(Card $card)
    {
        $this->card = $card;
    }

    public function render()
    {
        return view('livewire.player');
    }
}
```

When I click the refresh button [the player will disappear](https://cln.sh/cYFJJ2Gz).

**Comment by joshhanley** [comment]
@lucaguindani whenever another library (like React) interacts with the DOM, you need to use `wire:ignore` on the element to stop Livewire from wiping out the changes when it re-renders the page. See here in the V2 docs (not sure why it's missing in V3 docs) https://laravel-livewire.com/docs/2.x/alpine-js#ignoring-dom-changes

Hope this helps!

**Reply by lucaguindani** [reply]
Thanks for your answer @joshhanley ! It's indeed working with the [simple refresh button example](https://github.com/livewire/livewire/discussions/6680#discussioncomment-6923683). The React component is not removed from the view on refresh.

But with a [more complex Livewire component](https://github.com/livewire/livewire/discussions/6680#discussion-5597642) I have the same issue. In this component the player is not rendered when the page loads. We start by uploading a media, the component then shows the processing/transcoding message, and when the media is ready the React component (media player) should be displayed, but [it's not the case](https://cln.sh/TWfZ4P4N). I'm probably still missing something or should do it another way.

**Reply by lucaguindani** [reply]
@joshhanley do you know if it's possible to render a react component when a livewire view is refreshing (on poll) like I'm trying to accomplish in [the first example](https://github.com/livewire/livewire/discussions/6680#discussion-5597642) ? Since it's not rendered on page load but when the media transcoding is complete, do I have to render it another way ? Many thanks.
